src/type.h

```diff
@@ -127,7 +127,13 @@ class Type : public Traceable {
     bool IsNumericType() const { return IsFloatType() || IsIntType(); }
 
     /** Returns true if the type is any kind of dependent type. */
-    bool IsDependentType() const;
+    bool IsDependent() const;
+
+    /** Returns true if the type is count dependent (based on template non-type parameter). */
+    bool IsCountDependent() const;
```

Comment:
The term "count dependent" is not a standard thing. Hence it requires a definition. I would assume that's only a vector and array types with a count parameter dependent on the non-type template parameter (so things like non-type template parameters being used in other contexts are not included). But dichotomy "type dependent" vs "count dependent" suggests that expressions that use non-type template parameters are also "count dependent".

I.e.
```cpp
template <typename T, int N> int foo(T in) {
    int my_var = N - 1; // is "N-1" count dependent?
    T array[N];
    array[my_var] = 123;
    return array[my_var];
}
```

Reply:
Done